### DISCUSSION TITLE
Index files from harddrive

### DISCUSSION THREAD

**Discussion by blocus** [directus/directus#15972] [Feature Requests]
Hello, 

I'm using Directus on a self-hosted VPS, and I must upload some videos (10 files from 20Mb to 50Mb each). Uploading them from my device will take too long and I had 413 ERROR. So I connected to a backup server where I downloaded them with SCP. Unfortunately, I didn't find the files on Directus It was my fault. so I created new rows on the database and made a python script to rename the files and generate a new CSV file with this information `title;storage;filename_disk;filename_download;folder;filesize` And I uploaded the CSV file.

I was thinking about a feature to locate a file on the server's hard drive and move it to the upload folder and scan all metadata from it (size, width, height, duration, type, ...)

**Comment by rijkvanzanten** [comment]
Love that! Also hopeful that we can make files work more like the "database" in the two-way bound mirroring aspect. In addition to what you're mentioning above, it would also be nice to "introspect" the folder structure as well 🙂

**Reply by blocus** [reply]
I found on the contributing doc, there is the 80/20 Rule. I will start to create a PR about this feature and maybe it will be accepted on the vote

**Comment by blocus** [comment]
I forgot to include the script i wrote, It may helps someone

```python
from os import listdir, stat, rename
from os.path import isfile, join
import uuid
import csv

mypath = "."

files = []

for f in listdir(mypath):
    if isfile(join(mypath, f)) and "mp4" in f:
        print(f)

        file_stats = stat(f)
        id = str(uuid.uuid1())
        files.append(
            {
                "title": f.replace(".mp4", "").strip(),
                "storage": "local",
                "filename_disk": id + ".mp4",
                "filename_download": f,
                "folder": "53962c38-773e-43f1-82f0-b4d3a8e09051",
                "size": file_stats.st_size,
            }
        )

        rename(f, id + ".mp4")
file_info = ["title", "storage", "filename_disk", "filename_download", "folder", "size"]


with open("import.csv", "w") as csvfile:
    writer = csv.DictWriter(csvfile, fieldnames=file_info)
    writer.writeheader()
    writer.writerows(files)
```

**Reply by vanling** [reply]
I really really needed this right now :D

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
